firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/InAppMessageStreamManager.java

```diff
@@ -232,22 +243,37 @@ private boolean shouldIgnoreCache(String event) {
                       .defaultIfEmpty(CampaignImpressionList.getDefaultInstance())
                       .onErrorResumeNext(Maybe.just(CampaignImpressionList.getDefaultInstance()));
 
+              Maybe<InstanceIdResult> getIID = taskToMaybe(firebaseInstanceId.getInstanceId());
+
               Function<CampaignImpressionList, Maybe<FetchEligibleCampaignsResponse>> serviceFetch =
-                  impressions ->
-                      taskToMaybe(apiClient.getFiams(impressions))
-                          .doOnSuccess(
-                              resp ->
-                                  Logging.logi(
-                                      String.format(
-                                          Locale.US,
-                                          "Successfully fetched %d messages from backend",
-                                          resp.getMessagesList().size())))
-                          .doOnSuccess(
-                              resp -> impressionStorageClient.clearImpressions(resp).subscribe())
-                          .doOnSuccess(analyticsEventsManager::updateContextualTriggers)
-                          .doOnSuccess(testDeviceHelper::processCampaignFetch)
-                          .doOnError(e -> Logging.logw("Service fetch error: " + e.getMessage()))
-                          .onErrorResumeNext(Maybe.empty()); // Absorb service failures
+                  campaignImpressionList -> {
+                    if (!dataCollectionHelper.isAutomaticDataCollectionEnabled()) {
+                      Logging.logi(
+                          "Automatic data collection is disabled, not attempting campaign fetch from service.");
+                      return Maybe.just(cacheExpiringResponse());
+                    }
+
+                    // blocking get occurs on the IO thread because that's what we observeOn above
+                    return Maybe.fromCallable(getIID::blockingGet)
```

Comment:
You should be able to just return getIID here, which is already a Maybe

Reply:
ah but I need to do blocking get which turns it into a not maybe, if I don't do blocking get here the waiting ends up hitting the main thread below